cypress/support/bot.ts

```diff
@@ -63,6 +77,26 @@ declare global {
     }
 }
 
+Cypress.Commands.add("newMatrixClient", (
+    synapse: SynapseInstance,
+    { userId, accessToken, deviceId }: INewMatrixClientOptions,
+): Chainable<MatrixClient> => {
+    return cy.window({ log: false }).then(win => {
+        const cli = new win.matrixcs.MatrixClient({
+            baseUrl: synapse.baseUrl,
+            userId,
+            deviceId,
+            accessToken,
+            request,
+        });
+
+        cli.startClient();
+
+        return cli;
+    });
+});
+
+// TODO: Can we re-use `newMatrixClient` again here?
```

Comment:
TODO: Can we re-use `newMatrixClient` again here?

Here is how it looked before we merged `develop`, https://github.com/matrix-org/matrix-react-sdk/pull/8354/files/34ebd08e823430c35029617f090164196d543be8#diff-24813c28680122b1ec4f3a24a05dba300bbe727398976c2b7623866a57db08a6R72-R92

Reply:
Doesn't seem better with the extra layer. Doesn't seem to hurt to do it separate.

```js
Cypress.Commands.add("getBot", (synapse: SynapseInstance, opts: CreateBotOpts): Chainable<MatrixClient> => {
    opts = Object.assign({}, defaultCreateBotOptions, opts);
    const username = Cypress._.uniqueId("userId_");
    const password = Cypress._.uniqueId("password_");
    return cy.registerUser(synapse, username, password, opts.displayName).then(credentials => {
        return cy.window({ log: false }).then(win => {
            return cy.newMatrixClient(synapse, {
                baseUrl: synapse.baseUrl,
                userId: credentials.userId,
                deviceId: credentials.deviceId,
                accessToken: credentials.accessToken,
                request,
                store: new win.matrixcs.MemoryStore(),
                scheduler: new win.matrixcs.MatrixScheduler(),
                cryptoStore: new win.matrixcs.MemoryCryptoStore(),
            });
        }).then((cli) => {
            cy.window({ log: false }).then(win => {
                if (opts.autoAcceptInvites) {
                    cli.on(win.matrixcs.RoomMemberEvent.Membership, (event, member) => {
                        if (member.membership === "invite" && member.userId === cli.getUserId()) {
                            cli.joinRoom(member.roomId);
                        }
                    });
                }
            });

            return cy.wrap(
                cli.initCrypto()
                    .then(() => cli.setGlobalErrorOnUnknownDevices(false))
                    .then(() => cli.startClient())
                    .then(() => cli.bootstrapCrossSigning({
                        authUploadDeviceSigningKeys: async func => { await func({}); },
                    }))
                    .then(() => cli),
            );
        });
    });
});
```